chemprop/data/datasets.py

```diff
@@ -442,3 +442,19 @@ def normalize_inputs(
 
     def reset(self):
         return [dset.reset() for dset in self.datasets]
+
+    @property
+    def d_xd(self) -> list[int]:
+        return self.datasets[0].d_xd
+
+    @property
+    def d_vf(self) -> list[int]:
+        return sum(dset.d_vf for dset in self.datasets)
+
+    @property
+    def d_ef(self) -> list[int]:
+        return sum(dset.d_ef for dset in self.datasets)
+
+    @property
+    def d_vd(self) -> list[int]:
+        return sum(dset.d_vd for dset in self.datasets)
```

Comment:
Just a note that d_vf, d_ef, and d_vd for multicomponent aren't really defined because each monocomponent dataset runs through message passing separately. There is never a vector of length d_vf where d_vf is the sum of d_vf for each monocomponent dataset because the atom features for each monocomponent dataset are always kept separate. One example case is say two components share a message passing layer. Then they would have the same d_vf, but the multicomponent dataset would see d_vf as double that. 

But it doesn't matter because we only use this in the code to check in the CLI if it is greater than zero and in that case summing is the correct thing to do.